src/fight.cpp

```diff
@@ -395,17 +395,30 @@ void set_fighting(struct char_data * ch, struct char_data * vict, ...)
     }
   }
   if (!already_there) {
-    ch->next_fighting = combat_list;
-    combat_list = ch;
+    // Setting init to 0 here means new combat actually starts with the next global re-roll.
+    // This prevents arbitray length out-of-initiative states as new combats are initiated.
+    // Also need to avoid being first in the list to avoid setting off an early global re-roll.
+    GET_INIT_ROLL(ch) = 0;
```

Comment:
The implication of this is that if combat is already going on and you do `kill guard`, you'll sit there getting out-of-initiative warnings until the next combat reset, which is not a good experience. I see you've added a message about it below, but I'm not convinced that this is the best way to do it. Maybe roll their init as normal, then cap it to the highest init already in the combat list?

Reply:
Ah, that does sound like the better solution, I'll go ahead and do that (which also means the above re-location of the surprise flagging is no longer needed).

One thing I need to think about is the order of actions if both the player and their target roll higher initiative than the current cap. If I understand right, the attacker will be appended to the front of the list first, and then the defender, and order_list will retain that order. So if both exceed the cap, the defender will always act before the attacker, regardless of their actual rolls.